addons/point_of_sale/controllers/main.py

```diff
@@ -13,17 +14,16 @@ class PosController(http.Controller):
 
     @http.route('/pos/web', type='http', auth='user')
     def a(self, debug=False, **k):
-        cr, uid, context, session = request.cr, request.uid, request.context, request.session
+        session = request.session
 
         # if user not logged in, log him in
         if not session.uid:
             return login_redirect()
 
-        PosSession = request.registry['pos.session']
-        pos_session_ids = PosSession.search(cr, uid, [('state','=','opened'),('user_id','=',session.uid)], context=context)
+        pos_session_ids = request.env['pos.session'].search(
```

Comment:
search is not return ids. so declare variable like pos_session instead of pos_session_ids